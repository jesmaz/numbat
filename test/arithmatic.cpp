#include "interpretertest.hpp"

TEST_F (InterpreterTest, integer) {
	
	EXPECT_EQ (interpret ("1"), "1");
	EXPECT_EQ (interpret ("01"), "1");
	EXPECT_EQ (interpret ("10"), "10");
	EXPECT_EQ (interpret ("56"), "56");
	EXPECT_EQ (interpret ("0000000000012345"), "12345");
	
}

TEST_F (InterpreterTest, integer_addition) {
	
	EXPECT_EQ (interpret ("1 + 1"), "2");
	EXPECT_EQ (interpret ("1 + 1"), "2");
	
	EXPECT_EQ (interpret ("13 + 8"), "21");
	
	EXPECT_EQ (interpret ("1398231 + 3890278"), "5288509");
	
}

TEST_F (InterpreterTest, integer_subtraction) {
	
	EXPECT_EQ (interpret ("1 - 1"), "0");
	EXPECT_EQ (interpret ("2 - 1"), "1");
	EXPECT_EQ (interpret ("2 - 0"), "2");
	EXPECT_EQ (interpret ("0 - 2"), "-2");
	EXPECT_EQ (interpret ("4 - 2 - 3"), "-1");
	
}
