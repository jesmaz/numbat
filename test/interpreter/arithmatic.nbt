10
7849200011
12 + 30
90 + 10
50 - 40
10 - 24
2 * 16
10 * 0.5
8 / 2
1 / 2.0
1 + 2 * 3
10 * -4
7 / 2
7 / 2.0