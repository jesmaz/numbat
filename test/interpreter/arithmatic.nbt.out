10
7849200011
42
100
10
-14
32
5.000000
4
0.500000
7
-40
3
3.500000